converter_2.go

```
@@ -0,0 +1,17 @@
+package main
+
+import (
+	"fmt"
+)
+
+func myStrToInt2(str string) (int, error) {
+	var b int
+	x, err := fmt.Sscanf(str, "%d", &b)
+	if str == "" {
+		return 1, error(nil)
+	}
+	if x == 0 || err != nil {
+		return 2, error(nil)
```

Why did you return 2 here? This is not a code, that will be returned to the OS. This is just a value, that will be processed by the caller.
You should return 0 (default value for int) and error values:
`return 0, err`

---

Please return an error instead of special values

---

++